modules/apigee/main.tf

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  org_id       = try(google_apigee_organization.organization[0].id, "organizations/${var.project_id}")
+  envgroups    = coalesce(var.envgroups, {})
+  environments = coalesce(var.environments, {})
+  instances    = coalesce(var.instances, {})
+}
+
+resource "google_apigee_organization" "organization" {
+  count                                = var.organization == null ? 0 : 1
+  analytics_region                     = var.organization.analytics_region
+  project_id                           = var.project_id
+  authorized_network                   = var.organization.authorized_network
+  billing_type                         = var.organization.billing_type
+  runtime_type                         = var.organization.runtime_type
+  runtime_database_encryption_key_name = var.organization.database_encryption_key
+}
+
+resource "google_apigee_envgroup" "envgroups" {
+  for_each  = local.envgroups
+  name      = each.key
+  hostnames = each.value
+  org_id    = local.org_id
+}
+
+resource "google_apigee_environment" "environments" {
+  for_each        = local.environments
+  name            = each.key
+  display_name    = each.value.display_name
+  description     = each.value.description
+  deployment_type = each.value.deployment_type
+  api_proxy_type  = each.value.api_proxy_type
+  dynamic "node_config" {
+    for_each = try(each.value.node_config, null) != null ? [""] : []
+    content {
+      min_node_count               = node_config.min_node_count
+      max_node_count               = node_config.max_node_count
+      current_aggregate_node_count = node_config.current_aggregate_node_count
+    }
+  }
+  org_id = local.org_id
+}
+
+resource "google_apigee_envgroup_attachment" "envgroup_attachments" {
+  for_each = local.environments
+  envgroup_id = (lookup(local.envgroups, each.value.envgroup, null) == null ?
```

Comment:
try(local.envgroups[each.value.envgroup], google_apigee_envgroup.envgroups[each.value.envgroup].id)

Reply:
done, but it's not quite like what you suggested